rfcs/0194-flake-entrypoint.md

```diff
@@ -0,0 +1,389 @@
+- Make Nix itself more complex and opinionated
+- Reduce flexibility for framework authors to innovate
+- Still require users to write some Nix code
+- Ossify its whole domain due to the combination of Hyrum's law and the desire for reproducibility
+
+## Extend outputs to accept framework function
+
+Allow `outputs` to be set to a framework function directly,
+such as `outputs = flake-parts.lib.mkFlake;`.
+The framework function would receive similar parameters to `callFlakeEntrypoint`.
+However, this would require the input (such as `flake-parts` in this example) to be brought into scope,
+requiring that `outputs` is a function, and when `outputs` is a function,
+it should probably just be compatible with the status quo semantics for it.
+So that puts us back at square one.
+
+This would be less convenient but wouldn't require the `inputs.entrypoint` special case.
+However, it still requires writing the function invocation in every flake
+and doesn't enable the zero-Nix-code vision when combined with RFC 193.
+
+## Template-based code generation
+
+Use code generation tools (like `nix flake init`) to generate the boilerplate
+instead of removing it via abstraction.
+
+This approach keeps flakes explicit but:
+- Generated code still needs to be maintained and understood
+- Doesn't reduce the amount of code in repositories
+- Makes it harder to adopt framework improvements, as it requires regenerating,
+  and redoing any changes, flipping the supposed advantage of having the code
+  right under your fingers on its head.
+
+## Different entrypoint naming
+
+Instead of `inputs.entrypoint`, use a different name or location
+(e.g., `outputs.entrypoint`, `framework`, or a top-level `entrypoint` attribute).
+
+However, `inputs.entrypoint` aligns well with the inputs concept
+since the entrypoint is itself a dependency.
+The chosen approach keeps the interface and implementation simple.
+
+# Prior art
+[prior-art]: #prior-art
+
+## RFC 193 - TOML Flakes
+
+[RFC 193](https://github.com/NixOS/rfcs/pull/193) proposes using `flake.toml` for declarative input specifications.
+Combined with the entrypoint mechanism proposed here,
+flakes could be derived without *any* custom Nix wiring code,
+instead inferring outputs exclusively from the source tree and designated `inputs`.
+Expressions could be provided to adjust behavior on an as-needed basis only.
+
+The entrypoint mechanism can be implemented independently of TOML,
+but the two features work well together.
+
+## flake-parts, flake-utils-plus and blueprint
+
+Framework systems like flake-parts, flake-utils-plus and blueprint already provide abstractions over flake outputs.
+This proposal would enable these frameworks to be specified declaratively,
+in a standard way,
+without boilerplate.
+
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- What outcomes result from a prototype of this feature?
+- How should errors be reported when an entrypoint fails?
```

Comment:
A simple message `entrypoint "flake-parts"` failed with the above error.`, preceded by an error trace, seems like a workable solution.